authui/src/customSelect.ts

```diff
@@ -1,38 +1,160 @@
 import { Controller } from "@hotwired/stimulus";
+import { autoPlacement, computePosition, flip, shift } from "@floating-ui/dom";
 
 export interface SearchSelectOption {
+  triggerLabel?: string;
   label: string;
   value: string;
 }
 
 export class CustomSelectController extends Controller {
-  static targets = ["select", "itemTemplate"];
+  static targets = ["trigger", "dropdown", "options", "itemTemplate"];
   static values = {
     options: Array,
+    default: String,
   };
 
-  declare readonly selectTarget: HTMLSelectElement;
+  declare readonly triggerTarget: HTMLButtonElement;
+  declare readonly dropdownTarget: HTMLElement;
+  declare readonly optionsTarget: HTMLElement;
+  declare readonly itemTemplateTarget: HTMLTemplateElement;
 
   declare readonly optionsValue: SearchSelectOption[];
 
-  // Update select options when options change
+  _selectedItemIndex: number = -1;
+
+  connect(): void {
+    this.dropdownTarget.classList.add("hidden");
+    computePosition(this.triggerTarget, this.dropdownTarget, {
+      placement: "top",
+      middleware: [flip(), shift(), autoPlacement({ alignment: "start" })],
+    }).then(({ x, y }) => {
+      Object.assign(this.dropdownTarget.style, {
+        left: `${x}px`,
+        top: `${y}px`,
+      });
+    });
+
+    this.triggerTarget.addEventListener("keydown", this.handleKeyDown);
+    this.dropdownTarget.addEventListener("keydown", this.handleKeyDown);
+    document.addEventListener("click", this.handleClickOutside);
```

Comment:
Have you tried https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog ? Does it work?

Reply:
Couldn't get focus to work when using dialog, will try again after addressing other comments.

Reply:
Able to make dialog work with current country select. 

The dialog features used: `::backdrop`, `.show` should have ~95% on browser support.